subprojects/plugin/src/main/java/ca/coglinc/gradle/plugins/javacc/JavaccPlugin.java

```diff
@@ -60,7 +60,7 @@ private void addCompileJjdocTaskToProject(Project project, Configuration configu
     }
 
     private void addTaskToProject(Project project, Class<? extends AbstractJavaccTask> type, String name, String description, String group, Configuration configuration) {
-        Map<String, Object> options = new HashMap<String, Object>(2);
+        Map<String, Object> options = new HashMap<String, Object>(3);
```

Comment:
Below, three objects are put into `options`. Constructor of `HashMap` which takes an `int` is a bit confusing, and this change won't guarantee prevention of rehashing, as the description of the PR claims. The misleading constructor of `HashMap` is described in detail in <https://bugs.openjdk.org/browse/JDK-8284377>.

JDK-8284377 was released in Java 19, so it is not yet accessible to JavaCC due to max version limited to Java 8 (see README). A more correct way of calculating capacity for a pre-sized `HashMap` in Java versions before Java 19 is `numberOfElements / 0.75`, where 0.75 is the default load factor. See, for example, [change for JDK-8284377 in HostLocaleProviderAdapterImpl.java](https://github.com/openjdk/jdk/commit/87faa85c59e94d66c3c61d997eacdd2dbe5a1772#diff-5c96b8c01cf42bed243cf9c9e2eb20619477e88e3a77028c6f0c03a042ac16b3).

In this particular case, `3 / 0.75 == 4`, although other places in code of JDK also add `1` on top of that:

- [Character.java](https://github.com/openjdk/jdk/commit/87faa85c59e94d66c3c61d997eacdd2dbe5a1772#diff-0748178d01743c67f7a69bfd36a13068ddceed31da2e09c316ee9b7122063bc9):
   ```java
           private static Map<String, UnicodeBlock> map =
                new HashMap<>((int)(NUM_ENTITIES / 0.75f + 1.0f));
           [...]
                            aliases = new HashMap<>((int)(162 / 0.75f + 1.0f));
   ```
- [Class.java](https://github.com/openjdk/jdk/commit/87faa85c59e94d66c3c61d997eacdd2dbe5a1772#diff-a6270f4b50989abe733607c69038b2036306d13f77276af005d023b7fc57f1a2):
   ```java
               directory = new HashMap<>((int)(universe.length / 0.75f) + 1);
   ```

Reply:
@rybak how did you find this issue? (It's probably not worth changing again since the impact is negligible, though it's good to know about the discrepancy.)